pinot-broker/src/main/java/com/linkedin/pinot/broker/broker/helix/ClusterChangeMediator.java

```diff
@@ -52,49 +57,51 @@ public ClusterChangeMediator(HelixExternalViewBasedRouting helixExternalViewBase
     _deferredClusterUpdater = new Thread("Deferred cluster state updater") {
       @Override
       public void run() {
-        int lastExternalViewChangeCount = 0;
-        int lastInstanceConfigChangeCount = 0;
-
         while (true) {
-          // Check if we need to update the external view
-          final int currentExternalViewChangeCount = _externalViewChangeCount.get();
-          if (currentExternalViewChangeCount != lastExternalViewChangeCount) {
-            try {
-              _helixExternalViewBasedRouting.processExternalViewChange();
-              _lastExternalViewUpdateTime = System.currentTimeMillis();
-            } catch (Exception e) {
-              LOGGER.warn("Caught exception when processing external view change", e);
+          try {
+            // Wait for at least one update
+            Pair<UpdateType, Long> firstUpdate = _clusterChangeQueue.take();
+
+            // Take all other updates also present
+            List<Pair<UpdateType, Long>> allUpdates = new ArrayList<>();
+            allUpdates.add(firstUpdate);
+            _clusterChangeQueue.drainTo(allUpdates);
+
+            // Gather all update types
+            boolean externalViewUpdated = false;
+            boolean instanceConfigUpdated = false;
+
+            for (Pair<UpdateType, Long> update : allUpdates) {
+              if (update.getKey() == UpdateType.EXTERNAL_VIEW) {
+                externalViewUpdated = true;
+              } else if (update.getKey() == UpdateType.INSTANCE_CONFIG) {
+                instanceConfigUpdated = true;
+              }
             }
 
-            lastExternalViewChangeCount = currentExternalViewChangeCount;
-          }
-
-          // Check if we need to update the instance configs
-          final int currentInstanceConfigChangeCount = _instanceConfigChangeCount.get();
-          if (currentInstanceConfigChangeCount != lastInstanceConfigChangeCount) {
-            try {
-              _helixExternalViewBasedRouting.processInstanceConfigChange();
-              _lastInstanceConfigUpdateTime = System.currentTimeMillis();
-            } catch (Exception e) {
-              LOGGER.warn("Caught exception when processing instance config change", e);
+            if (externalViewUpdated) {
+              try {
+                _helixExternalViewBasedRouting.processExternalViewChange();
```

Comment:
Do you want to add or keep track of time taken to process here, or is that done some place else?